test/CadlRanchProjects.Tests/type-property-optionality.cs

```diff
@@ -590,5 +589,90 @@ public void PlainTime()
             var output = PlainTimeProperty.DeserializePlainTimeProperty(document.RootElement);
             Assert.AreEqual(PlainTimeData, output.Property);
         }
+
+        [Test]
+        public void NotRequiredNullablePropertiesOmitedByDefault()
+        {
+            var inputModel = new PlainTimeProperty();
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.False(element.TryGetProperty("property", out _));
+        }
+
+        [Test]
+        public void NotRequiredPropertiesDeserializedWithNullsWhenUndefined()
+        {
+            var model = PlainTimeProperty.DeserializePlainTimeProperty(JsonDocument.Parse("{}").RootElement);
+            Assert.Null(model.Property);
+        }
+
+        [Test]
+        public void NotRequiredPropertiesDeserializedWithValues()
+        {
+            var model = StringProperty.DeserializeStringProperty(JsonDocument.Parse("{\"property\": \"2\"}").RootElement);
+            Assert.AreEqual("2", model.Property);
+        }
+
+        [Test]
+        public void InputModelDoesntSerializeOptionalCollections()
+        {
+            var inputModel = new CollectionsByteProperty();
+
+            // Perform non mutating operations
+            _ = inputModel.Property.Count;
+            _ = inputModel.Property.IsReadOnly;
+            _ = inputModel.Property.Remove(BinaryData.FromString("a"));
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+
+            Assert.False(element.TryGetProperty("property", out _));
+        }
+
+        [Test]
+        public void InputModelSerializeOptionalCollectionAfterMutation()
```

Comment:
same here. The subject is the property not the model

Reply:
fixed